components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/ha/HACoordinationSourceHandler.java

```diff
@@ -164,15 +173,16 @@ private void sendEventsToPassiveNode(Event event) {
         }
     }
 
-    private void sendEventsToPassiveNode(Event[] events) {
+    private void sendEventsToPassiveNode(Event[] events, String[] transportSyncProperties) {
         EventSyncConnection eventSyncConnection = getTCPNettyClient();
         ByteBuffer messageBuffer = null;
         if (eventSyncConnection != null) {
             QueuedEvent[] queuedEvents = new QueuedEvent[events.length];
             int i = 0;
             for (Event event : events) {
                 QueuedEvent queuedEvent = new QueuedEvent(siddhiAppName, sourceHandlerElementId, sequenceIDGenerator
-                        .incrementAndGet(), event);
+                        .incrementAndGet(), event,
+                        transportSyncProperties != null ? new String[]{transportSyncProperties[i]} : null);
```

Comment:
This is wrong. 
Here we can get multiple events but have a common set of transport properties. 
So one option is to add all the properties to the first message only. 
And make sure not to call the updateTransportSyncProperties() when transport properties are null

Reply:
@suhothayan If we take kafka source, when publishing events i'm putting following transport properties, "topic:topicVal,partition:partitionVal,offset:offsetVal", So each event will send these with different offsets. so this data can be common in some other source  but also different in other sources like kafka case. wdyt ?

Reply:
Fixed